Arelle/ixbrl-viewer
# Multi document support

#### Reason for change

The viewer needs to support viewing filings comprising multiple separate Inline XBRL Documents or Document Sets.

#### Description of change

The Python plugin will now convert all loaded reports into the viewer.  Each report document will be loaded into a separate tab in the viewer.  Note that a source report may be a Inline XBRL Document Set, so each report may contribute multiple tabs to the viewer.

Facts from all reports will be shown in the fact inspector, and search will find facts from all reports.

#### Steps to Test

Generate a viewer, specifying multiple Inline XBRL source documents.  For example, the 5.2.5 series tests from the [EFM conformance suite](https://www.sec.gov/files/edgar/efm-67-230918.zip):

```
cd conf/525-ix-syntax/efm/24-multixds/mi-01-pri-gd-att-gd
mkdir out
python3 $ARELLE/arelleCmdLine.py --plugin  iXBRLViewerPlugin  -v -f '[{"file":"eelo00001gd-20340331.htm","attachmentDocumentType":"10-K"},{"file":"eelo00002gd-20340331.htm","attachmentDocumentType":"EX-26"}]' --save-viewer out
```

Now open the resulting files via a web browser.  The result should appear much like an Inline XBRL Document Set with two tabs, but these are actually separate reports, with a separate DTS.

It is also possible to load multiple reports by putting them in a ZIP, or simply listing the filenames separated by `|`.

Note that the fact IDs (visible in the browser location bar) are now prefixed with `0-` or `1-` for the different documents.

There are significant changes to the structure of the JSON data.  These are described in `ARCHITECTURE.md`.  Some of these changes are designed to cope with multiple target documents, but this is not yet implemented. The viewer should render reports using the old JSON format correctly.  


**review**:
@Arelle/arelle
@paulwarren-wk
